### PR TITLE
Add sort commands

### PR DESCRIPTION
**Pull request by yusiangeng**
Fixes https://github.com/AY2223S1-CS2103T-F12-2/tp/issues/76

## Changes

In `model/ModelManager.java`, added a `SortedList<Task> sortedTasks` which is wrapped by `FilteredList<Task> filteredTasks`.

## Usage

Sort by deadline (if no argument provided, defaults to ascending order):

```bash
# ascending order
sortByDeadline 
sortByDeadline o/asc
sortByDeadline o/a

# descending order
sortByDeadline o/desc
sortByDeadline o/d
```

Sort by priority (if no argument provided, defaults to descending order):

```bash
# descending order
sortByPriority 
sortByPriority o/desc
sortByPriority o/d

# ascending order
sortByPriority o/asc
sortByPriority o/a
```

## Demo

https://user-images.githubusercontent.com/96862218/197450861-77c18cbd-fb42-45b3-8715-bbe4f27c8300.mp4

### CODE REVIEW

File: src/main/java/seedu/address/model/Model.java [hunk 1]
```diff
@@ -129,4 +130,11 @@ public interface Model {
      * @throws NullPointerException if {@code predicate} is null.
      */
     void updateFilteredTaskList(Predicate<Task> predicate);
+
+    /**
+     * Updates the sort order of the filtered task list using the given {@code comparator}.
+     *
+     * @throws NullPointerException if {@code comparator} is null.
+     */
+    void updateFilteredTaskList(Comparator<Task> comparator);
```
Comment (ALWHgithub) [inline src/main/java/seedu/address/model/Model.java:139]: Maybe naming it updateSortedTaskComparator or updateSortingCriteria would be better ?

**Review by ALWHgithub — COMMENTED** [review]
Looks mostly fine, other than a few minor points :)

File: src/main/java/seedu/address/model/Model.java (hunk 1 above)
Comment (yusiangeng) [reply src/main/java/seedu/address/model/Model.java]: updated

File: src/main/java/seedu/address/logic/parser/SortTaskDeadlineParser.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                break;
+            default:
+                throw new ParseException(
+                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, SortTaskDeadlineCommand.MESSAGE_USAGE));
+            }
+        }
+
+        return new SortTaskDeadlineCommand(comparator);
+    }
+
```
[the remark this answers is not in the document]
Comment (yusiangeng) [reply src/main/java/seedu/address/logic/parser/SortTaskDeadlineParser.java:53]: i tried but i dont know how to do this easily :(

**Pull request merged**